src/bun.js/api/YAMLObject.zig

```diff
@@ -925,7 +925,15 @@ pub fn parse(
     const root = bun.interchange.yaml.YAML.parse(source, &log, arena.allocator()) catch |err| return switch (err) {
         error.OutOfMemory => |oom| oom,
         error.StackOverflow => global.throwStackOverflow(),
-        else => global.throwValue(try log.toJS(global, bun.default_allocator, "Failed to parse YAML")),
+        else => {
+            // Throw a SyntaxError instead of BuildMessage to match JSON.parse behavior
+            if (log.msgs.items.len > 0) {
+                const first_msg = log.msgs.items[0];
+                const error_text = first_msg.data.text;
+                return global.throwValue(global.createSyntaxErrorInstance("YAML Parse error: {s}", .{error_text}));
+            }
+            return global.throwValue(global.createSyntaxErrorInstance("Failed to parse YAML", .{}));
```

Comment:
```suggestion
            return global.throwValue(global.createSyntaxErrorInstance("YAML Parse error: Unable to parse YAML string", .{}));
```
matching JSC JSON parse error message